main.go

```diff
@@ -175,6 +175,33 @@ func doesDaemonSetExist(daemonSetName, namespace string) bool {
 	return err == nil
 }
 
+func DaemonSetReady(daemonSetName, namespace, image string) bool {
+	const hoursPerWeek = 168
+
+	// The daemon set will be considered not ready if it does not exist
+	ds, err := daemonsetClient.K8sClient.AppsV1().DaemonSets(namespace).Get(context.TODO(), daemonSetName, metav1.GetOptions{})
+	if err != nil {
+		logrus.Infof("could not get daemon set %s, err=%s", daemonSetName, err.Error())
+		return false
+	}
+
+	// Or if it's been running for more than a week
+	if time.Since(ds.CreationTimestamp.Time).Hours() > hoursPerWeek {
+		return false
+	}
+
+	// Or if the containers images do not match the desired one
+	containers := ds.Spec.Template.Spec.Containers
+	for i := 0; i < len(containers); i++ {
```

Comment:
I don't think we need to iterate through the containers, as this daemonset has only one: the public function `CreateDaemonSet`  admits only one container image. So you can do this:
```
if containers[0].Image != image {
  return false
}
```